test-core/src/test/groovy/io/micronaut/starter/core/test/feature/github/workflow/DockerRegistryWorkflowSpec.groovy

```diff
@@ -0,0 +1,113 @@
+package io.micronaut.starter.core.test.feature.github.workflow
+
+import io.micronaut.starter.feature.github.workflows.Secret
+import io.micronaut.starter.feature.github.workflows.docker.DockerRegistryWorkflow
+import io.micronaut.starter.feature.github.workflows.docker.GraalVMDockerRegistryWorkflow
+import io.micronaut.starter.options.BuildTool
+import io.micronaut.starter.options.Language
+import io.micronaut.starter.test.github.WorkflowSpec
+import io.micronaut.starter.util.NameUtils
+import spock.lang.Requires
+import spock.lang.Shared
+
+/**
+ * This integration spec contains tests for both {@link DockerRegistryWorkflow} and {@link GraalVMDockerRegistryWorkflow}.
+ * To run this spec env variable required to be in env which are afterwards automatically created as GitHub Secrets:
+ * <ul>
+ *     <li>DOCKER_USERNAME - Username used to authenticate to Docker registry</li>
+ *     <li>DOCKER_PASSWORD - Password used to authenticate to Docker registry</li>
+ *     <li>DOCKER_REGISTRY_URL - Docker registry url to authenticate aga repository path.</li>
+ *     <li>DOCKER_ORGANIZATION - Docker organisation repository path.</li>
+ * </ul>
+ */
+@Requires({System.getenv().containsKey(GH_TOKEN) && System.getenv().containsKey(DOCKER_USERNAME) \
```

Comment:
It would be great to configure these secrets in GitHub project in order to run this test at least against dockerhub:
- the repositories have to exists in docker hub
- ! ideally for GH_TOKEN to create some system account b/c the token needs to have permission to delete projects what is quite dangerous operation.

Reply:
We can have a different token than `GH_TOKEN`. Can you create an issue so that we don't forget? I think it's worth to test it, but I would not hold this PR because of that.